torch/_functorch/vmap.py

```diff
@@ -227,34 +229,38 @@ def lazy_load_decompositions():
     global DECOMPOSITIONS_LOADED
     if DECOMPOSITIONS_LOADED:
```

Comment:
This needs to be removed.

Reply:
This shortcut can skip unnecessary lock acquirements. This will speed up subsequent function calls.

Once the `DECOMPOSITIONS_LOADED` variable is set `True`. It will never be set to `False`. So we can have this before the lock context environment.

Reply:
But then this thread will assume the state is initialized while it might not be true and you're going to fail to find the kernels you expect to be there.

Reply:
> You would need another flag at the end of the initialization if you wanted to do that.

Thanks! You are right. I moved the flag assignment to the end of the context.